src/main/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1749,4 +1790,33 @@ public int scrub(boolean disableSnapshot, boolean skipCorrupted, boolean checkDa
         APIClient.set_query_param(queryParams, "cf", APIClient.join(columnFamilies));
         return client.getIntValue("/storage_service/keyspace_scrub/" + keyspaceName, queryParams);
     }
+
+    @Override
+    public CompositeData getToppartitions(String sampler, List<String> keyspaceFilters, List<String> tableFilters, int duration, int capacity, int count) throws OpenDataException {
```

Comment:
I've comment on the other PR, you can just keep keyspaceFilters as is for the entire way from nodetool to the API

Reply:
I don't see any comments about this on the other PRs. What do you mean by keeping those "as is"?

Reply:
@amnonh ping

Reply:
In the nodetool/nodeprobe side you take a string and split it into an array, then here you take that array and makes it a string, you can just pass it as it is

Reply:
It seems to me that it's more elegant and universal expecting a list, not a string in the jmx endpoint. List is self explanatory, while accepting a specifically separated string requires some form of conveying that information, like documentation or whatever. Also, things outside of jmx do not really need to know how the APIClient inside joins lists for queries - was it to ever change, we then would have to change it also in nodetool and every other place the exposed jmx endpoint is being called. The lists shouldn't be long so performance is not an issue either.

That's my reasoning behind this, but you have the final word. If you prefer it all to just pass the string around, I'll change it.